test/auto_parallel/pir/test_static_pir_program.py

```diff
@@ -0,0 +1,127 @@
+                input = paddle.static.data(
+                    name='input',
+                    shape=[BATCH_SIZE, SEQ_LEN, HIDDEN_SIZE],
+                )
+                w0 = paddle.pir.core.create_parameter(
+                    dtype="float32",
+                    shape=[HIDDEN_SIZE, HIDDEN_SIZE],
+                    name="w0",
+                    initializer=paddle.nn.initializer.Uniform(),
+                )
+                w1 = paddle.pir.core.create_parameter(
+                    dtype="float32",
+                    shape=[HIDDEN_SIZE, HIDDEN_SIZE],
+                    name="w1",
+                    initializer=paddle.nn.initializer.Uniform(),
+                )
+                self.assertTrue(input.is_dense_tensor_type())
+                self.assertTrue(w0.is_dense_tensor_type())
+
+                dist_input = dist.shard_tensor(input, mesh, [dist.Replicate()])
+                dist_w0 = dist.shard_tensor(w0, mesh, [dist.Shard(0)])
+                dist_w1 = dist.shard_tensor(w1, mesh, [dist.Shard(1)])
+
+        self.assertTrue(main_program.num_ops() == 6)
+
+        self.assertFalse(input.use_empty())
+        self.assertFalse(w0.use_empty())
+        self.assertFalse(w1.use_empty())
+
+        self.assertTrue(dist_input.use_empty())
+        self.assertTrue(dist_w0.use_empty())
+        self.assertTrue(dist_w1.use_empty())
+
+        self.assertTrue(w0.is_dense_tensor_type())
+        self.assertTrue(w1.is_dense_tensor_type())
+        self.assertTrue(input.is_dense_tensor_type())
+
+        # check dist type
+        self.assertTrue(dist_input.is_dist_dense_tensor_type())
+        self.assertTrue(dist_w0.is_dist_dense_tensor_type())
+        self.assertTrue(dist_w1.is_dist_dense_tensor_type())
+
+        # check global shape
+        self.assertTrue(dist_input.shape == [BATCH_SIZE, SEQ_LEN, HIDDEN_SIZE])
+        self.assertTrue(dist_w0.shape == [HIDDEN_SIZE, HIDDEN_SIZE])
+        self.assertTrue(dist_w1.shape == [HIDDEN_SIZE, HIDDEN_SIZE])
+        # check local shape
+        self.assertTrue(
+            dist_input._local_shape == dist_input.shape
+        )  # replicated, local = global
+        self.assertTrue(
+            dist_w0._local_shape == [HIDDEN_SIZE // MP_SIZE, HIDDEN_SIZE]
+        )  # sharded, local != global, sharded by mesh size
+        self.assertTrue(
+            dist_w1._local_shape == [HIDDEN_SIZE, HIDDEN_SIZE // MP_SIZE]
+        )  # sharded, local != global, sharded by mesh size
+
+        # check op dist_attr
+        self.assertFalse(input.get_defining_op().has_attr("op_dist_attr"))
+        self.assertFalse(w0.get_defining_op().has_attr("op_dist_attr"))
+        self.assertFalse(w1.get_defining_op().has_attr("op_dist_attr"))
+
+        dist_input_op = dist_input.get_defining_op()
+        self.assertTrue(dist_input_op.has_attr("op_dist_attr"))
+        self.assertTrue(dist_input_op.process_mesh == mesh)
+        self.assertTrue(dist_input_op.num_operand_dist_attrs() == 0)
```

Comment:
Not important. u can use assertEqual / assertNotEqual / assertTrue / assertFalse, etc, instead of `assertTrue` for convinience